File: Data Structures & Algorithms/daily-temperatures/submission-1.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] result = new int[temperatures.length];
        Deque<int[]> stack = new ArrayDeque <>();

        for(int i = 0; i < temperatures.length; i++) {
            int t = temperatures[i];
            while((!stack.isEmpty()) && (t > stack.peek()[0])) {
                int[] top = stack.pop();
                int stackInd = top[1];
                result[stackInd] = i - stackInd;
            } 
            
            stack.push(new int[]{t, i});
        }

        return result;
    }
}

File: Data Structures & Algorithms/minimum-stack/submission-1.py
class MinStack:

    def __init__(self):

        self.stack = []

    def push(self, val: int) -> None:
        self.stack.append(val)

    def pop(self) -> None:
        self.stack.pop()

    def top(self) -> int:
        n = len(self.stack)
        return self.stack[n-1]

    def getMin(self) -> int:
        return min(self.stack)
